parsers/CA_NS.py

```diff
@@ -1,213 +1,178 @@
 #!/usr/bin/env python3
-
-# The datetime library is used to handle datetimes
+import time
 from datetime import datetime, timezone
 from logging import Logger, getLogger
+from typing import Any
 
 from requests import Session
 
+from electricitymap.contrib.config import ZoneKey
+from electricitymap.contrib.lib.models.event_lists import (
+    ExchangeList,
+    ProductionBreakdownList,
+)
+from electricitymap.contrib.lib.models.events import ProductionMix
+from parsers.lib.exceptions import ParserException
+
+LOAD_URL = "https://www.nspower.ca/library/CurrentLoad/CurrentLoad.json"
+MIX_URL = "https://www.nspower.ca/library/CurrentLoad/CurrentMix.json"
+PARSER = "CA_NS.py"
+SOURCE = "nspower.ca"
+ZONE_KEY = ZoneKey("CA-NS")
+
+
+def _get_ns_info(
+    session: Session, logger: Logger
+) -> tuple[ExchangeList, ProductionBreakdownList]:
+    # Request data from the endpoints until the timestamps between the two
+    # arrays agree or the retry limit is reached.
+    for _ in range(3):
+        base_loads = session.get(LOAD_URL).json()  # Base loads in MW
+        mixes = session.get(MIX_URL).json()  # Electricity mix breakdowns in %
+        if all(
+            base_load["datetime"] == mix["datetime"]
+            for base_load, mix in zip(base_loads, mixes)
+        ):
+            break
+        time.sleep(2)
```

Comment:
While this is a smart move on it's own I'm not sure it works well with the backend as it already has it's own retry logic.

I have experienced this in another parser where it is the same 99.99% but when we refetched realy data one of the lists had a few hours more data which caused it to fail. By adding a fallback loop we avoid scenarios like this.